doc/ot_api_doc.h

```diff
@@ -191,3 +191,59 @@
  * @}
  *
  */
+/**
+ * @defgroup config                   Config Variables
```

Comment:
After testing with the updated script, can you move this config defgroup to a separate file entirely.

So `doc/ot_api_doc.h` would remain as is for the API reference.

Then there'd be a new one called `doc/ot_config_doc.h` with just this config defgroup.  Please include the same front matter the top (copyright, license, file/brief comment).  Then we'll use that one in our script, otherwise it will try to create pages for the other stuff, which will be blank.